### PR TITLE
Post clustering functionality

### PR DESCRIPTION
**Pull request by ngreenwald**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**
Adds in logic to annotate the cell table following pixel clustering, as well as detailed instructions for how to use Mantis. 

**How did you implement your changes**
A notebook with two parts. The first allows the user to select subsets of cells, give them new names, and then create a mantis directory to examine them. The second allows the user to identify specific thresholds for functional markers. 

**Remaining issues**
For people who haven't used Mantis, is this clear enough guidance for what they need to do? Are there places where the bare code in the notebook is going to lead to problems? Given that the user is directly manipulating the pandas df there are some inelegant lines still in there, but I couldn't think of a way to avoid it that wasn't super over-engineered.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,17 +84,47 @@ mantis_project
...
+└── marker_counts.csv
+
 ```
 
+### Loading image-specific files
+In addition to the images, there are additional files in the directory structure which can be read into mantis.  
+
+`cell_segmentation`: This file contains the predicted segmentation for each cell in the image, and allows mantis to identify individual cells.
+
+`population_pixel_mask`: This file maps the individual pixel clusters generated by Pixie in the [pixel clustering notebook](link) to the image data.
```
Comment (alex-l-kong) [inline README.md:104]: Add the link to the pixel clustering notebook.

File: ark/phenotyping/post_cluster_utils.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        os.makedirs(mask_dir)
+
+    # create small df compatible with FOV function
+    small_table = cell_table.loc[:, [pop_col, 'label', 'fov']]
+
+    # generate unique numeric value for each population
+    unique_pops = small_table[pop_col].unique()
+    small_table['pop_vals'] = small_table[pop_col].replace(to_replace=unique_pops,
+                                                           value=list(
+                                                               range(1, len(unique_pops) + 1)))
```
Comment (alex-l-kong) [inline ark/phenotyping/post_cluster_utils.py:65]:
```suggestion
    small_table['pop_vals'] = pd.factorize(small_table['pop_vals'].tolist())[0]
```

File: ark/phenotyping/post_cluster_utils.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        ax[i].hist(plot_vals, 50, density=True, facecolor='g', alpha=0.75, range=(0, x_max))
+        ax[i].set_title("Distribution of {} in {}".format(marker, populations[i]))
+
+        if threshold is not None:
+            ax[i].axvline(x=threshold)
+    plt.tight_layout()
+
+
+def create_mantis_project(cell_table, fovs, seg_dir, pop_col, mask_dir, image_dir, mantis_dir):
+    """Creates masks with the updated cell labels"""
```
Comment (alex-l-kong) [inline ark/phenotyping/post_cluster_utils.py:53]: Needs a docstring.

**Review by alex-l-kong — CHANGES_REQUESTED** [review]
Mostly small comments.

File: templates_ark/Clustering_cleanup.ipynb [hunk 4]
```diff
@@ -0,0 +1,395 @@
+{
```
Comment (alex-l-kong) [inline templates_ark/Clustering_cleanup.ipynb:1]: <p>Should we add an example visualization for this one?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/658/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (alex-l-kong) [inline templates_ark/Clustering_cleanup.ipynb:1]: <p>I'm assuming these links are being added in later?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/658/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (ngreenwald) [reply templates_ark/Clustering_cleanup.ipynb:1]: <p>Yeah agreed</p>

**Comment by ngreenwald** [comment]
@cliu72 @efrancis28 let me know if this makes sense for a post-clustering workflow, any other changes, etc

**Comment by cliu72** [comment]
This looks good to me.

**Pull request merged**